### PR TITLE
Update api docs for MeshConfig

### LINKED ISSUE #119: Update API docs to reflect addition of feature flags to the MeshConfig CRD
**Please describe what should be documented**

Update API reference docs for the MeshConfig custom API defined in `/pkg/apis/config/v1alpha1/`. The reference doc needs to include the featureFlag object.

### PR DESCRIPTION
**Pull request by jaellio**
Updates API reference docs for the MeshConfigs. Adds ClusterSpec and FeatureFlags.

Resolves #119

Signed-off-by: Jackie Elliott <jaellio@microsoft.com>

### CODE REVIEW

File: content/docs/apidocs/config/v1alpha1.md [hunk 1]
```diff
@@ -45,6 +45,46 @@ string
 </tr>
 </tbody>
 </table>
+<h3 id="config.openservicemesh.io/v1alpha1.ClusterSpec">ClusterSpec
+</h3>
+<p>
+(<em>Appears on:</em><a href="#config.openservicemesh.io/v1alpha1.MultiClusterServiceSpec">MultiClusterServiceSpec</a>)
```
Comment (shashankram) [inline content/docs/apidocs/config/v1alpha1.md:51]: I am inclined to not document the multicluster spec yet, given it's not ready and documenting it on the website is going to cause confusion.

Can we remove the multicluster def?

**Review by shashankram — COMMENTED** [review]
Thanks @jaellio. I made another change to the MeshConfig CRD in `main`, would you mind regenerating the doc from the HEAD of main.

File: content/docs/apidocs/config/v1alpha1.md (hunk 1 above)
Comment (snehachhabria) [reply content/docs/apidocs/config/v1alpha1.md:51]: +1

Comment (jaellio) [reply content/docs/apidocs/config/v1alpha1.md]: Good point, thanks!

**Review by bridgetkromhout — APPROVED** [review]
Thanks!

**Pull request merged**